interop-testing/src/main/java/io/grpc/testing/integration/GrpclbFallbackTestClient.java

```diff
@@ -0,0 +1,302 @@
+      throw new AssertionError(
+          "This test currently only supports "
+          + "--custom_credentials_type=compute_engine_channel_creds.");
+    }
+    ComputeEngineChannelBuilder builder = ComputeEngineChannelBuilder.forTarget(serverUri);
+    builder.keepAliveTime(3600, TimeUnit.SECONDS);
+    builder.keepAliveTimeout(20, TimeUnit.SECONDS);
+    return builder.build();
+  }
+
+  void setUp() {
+    channel = createChannel();
+    blockingStub = TestServiceGrpc.newBlockingStub(channel);
+  }
+
+  private void tearDown() {
+    try {
+      channel.shutdownNow();
+      channel.awaitTermination(1, TimeUnit.SECONDS);
+    } catch (Exception ex) {
+      throw new RuntimeException(ex);
+    }
+  }
+
+  private String streamToString(InputStream inputStream) throws Exception {
+    BufferedReader br = new BufferedReader(
+        new InputStreamReader(inputStream, UTF_8));
+    String out = "";
+    String line;
+    while ((line = br.readLine()) != null) {
+      out += line;
+    }
+    return out;
+  }
+
+  private void runShellCmd(String cmd) throws Exception {
+    logger.info("Run shell command: " + cmd);
+    Process process = Runtime.getRuntime().exec(new String[]{"bash", "-c", cmd});
+    int exitCode = process.waitFor();
+    logger.info("Shell command stdout: " + streamToString(process.getInputStream()));
+    logger.info("Shell command stderr: " + streamToString(process.getErrorStream()));
+    logger.info("Shell rommand exit code: " + exitCode);
+    assertEquals(0, exitCode);
+  }
+
+  private enum RpcMode {
+    FailFast,
+    WaitForReady,
+  }
+
+  private GrpclbRouteType doRpcAndGetPath(
+      TestServiceGrpc.TestServiceBlockingStub blockingStub,
+      int deadlineSeconds,
+      RpcMode rpcMode) {
+    logger.info("doRpcAndGetPath deadlineSeconds: " + deadlineSeconds + " rpcMode: "
+        + rpcMode);
+    final SimpleRequest request = SimpleRequest.newBuilder()
+        .setFillGrpclbRouteType(true)
+        .build();
+    GrpclbRouteType result = GrpclbRouteType.GRPCLB_ROUTE_TYPE_UNKNOWN;
+    try {
+      SimpleResponse response = null;
+      if (rpcMode == RpcMode.WaitForReady) {
+        response = blockingStub
+          .withDeadlineAfter(deadlineSeconds, TimeUnit.SECONDS)
+          .withWaitForReady()
+          .unaryCall(request);
+      } else {
+        response = blockingStub
+          .withDeadlineAfter(deadlineSeconds, TimeUnit.SECONDS)
+          .unaryCall(request);
+      }
+      result = response.getGrpclbRouteType();
+    } catch (StatusRuntimeException ex) {
+      logger.warning("doRpcAndGetPath failed. Status: " + ex);
+    }
+    if (result == GrpclbRouteType.GRPCLB_ROUTE_TYPE_FALLBACK
+        || result == GrpclbRouteType.GRPCLB_ROUTE_TYPE_BACKEND) {
+      throw new AssertionError("Received invalid LB route type. This suggests "
+          + "that the server hasn't implemented this test correctly.");
+    }
+    logger.info("doRpcAndGetPath. GrpclbRouteType result: " + result);
```

Comment:
You may want to log this before the if-block above.

Reply:
Good point, done